bot/seasons/easter/easter_riddle.py

```diff
@@ -0,0 +1,97 @@
+import asyncio
+import logging
+import random
+from json import load
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.constants import Colours
+
+log = logging.getLogger(__name__)
+
+with Path("bot/resources/easter/easter_riddle.json").open("r", encoding="utf8") as f:
+    RIDDLE_QUESTIONS = load(f)
+
+TIMELIMIT = 10
+
+
+class EasterRiddle(commands.Cog):
+    """This cog contains the command for the Easter quiz!"""
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.winners = []
+        self.correct = ""
+        self.current_channel = None
+
+    @commands.command(aliases=["riddlemethis", "riddleme"])
+    async def riddle(self, ctx):
+        """
+        Gives a random riddle, then provides 2 hints at certain intervals before revealing the answer.
+
+        The duration of the hint interval can be configured by changing the TIMELIMIT constant in this file.
+        """
+        if not self.current_channel:
```

Comment:
Instead of having this large body of code inside the if statement, you could turn the else condition into a return and have the body be executed only if the code did not return, i.e.

```py
if not channel:
    lots
    of
    code
else:
    await send(error)
```
turns into
```py
if channel:
    return await send(error)

lots
of
code
```
This way it is easier to follow the logic of the if statement as the else statement is not hidden by the large body inside the if statement. The function stays flatter, too.